### PR TITLE
:safety_vest: Added commit message validation

### PR DESCRIPTION
**Pull request by axel-eck**
I added Husky and made a script to check the commit message.

After noticing many commit messages not respecting the gitmoji norm which this repository follow, I decided to make a script to enforce this rule.

> [!WARNING]
> You should run `bun add --dev husky && bunx husky init`
> You may end up with a script in `.husky/pre-commit` you must remove this script if it appears

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  echo "Allowed emojis with their meanings are available at https://gitmoji.dev/"
+  echo "Example: \":art: Improve the design\""
+  echo "-"
+  echo "Your commit message was:"
+  echo "$message"
+  echo "-"
+  echo "For more information, check script in .husky/commit-msg"
+  echo "-"
+  exit 1
+fi
```
Comment (Marius-P1) [inline .husky/commit-msg]:
```suggestion
fi

```

**Review by 6im0n — APPROVED** [review]
:+1:

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by landryarki — APPROVED** [review]
if you want :+1:

**Pull request merged**